### PR TITLE
Added Look and Feel Suport for Map Creator

### PR DESCRIPTION
**Pull request by RoiEXLab**
One thing I realized... the Map creator is accepting arguments on startup, but the class firing the start of the map Crator passes no  arguments -> the only way passing those arguments is by manually starting the Map Creator class via command line (once the final jar file is built) which is not likely to be done by an user.

If you agree, I could simply remove this argument.

---
This change is [Reviewable](https://reviewable.io/reviews/triplea-game/triplea/621)

### CODE REVIEW

File: src/util/image/MapCreator.java [hunk 1]
```diff
@@ -66,6 +69,13 @@
   }
 
   public static void main(final String[] args) {
+      try {
+        UIManager.setLookAndFeel(UIManager.getSystemLookAndFeelClassName());//Fallback Look and Feel
+      } catch (ClassNotFoundException | InstantiationException | IllegalAccessException
+          | UnsupportedLookAndFeelException e) {
+        ClientLogger.logError("Default Look and Feel could not be applied", e);
```
Comment (DanVanAtta) [inline src/util/image/MapCreator.java]: This is a warning, not an error. Also no need for a full stack trace in this instance. Can do:
` ClientLogger.logError("Default Look and Feel could not be applied: "+ e.getMessage());`

File: src/util/image/MapCreator.java [hunk 2]
```diff
@@ -544,7 +554,20 @@ private static String getValue(final String arg) {
 
   private static void handleCommandLineArgs(final String[] args) {
     final String[] properties = getProperties();
-    if (args.length == 1) {
+    boolean laf = false;
```
Comment (DanVanAtta) [inline src/util/image/MapCreator.java]: Really pls spell out variable names: `laf`
I *did* not know what this stood for, had to search through the code to figure it out.
Also, it does not appear this variable is being used.

Comment (DanVanAtta) [inline src/util/image/MapCreator.java]: Does appear to be used actually, but it's not very clean. One if statement sets the value, then another uses it - a bit hard to follow.

File: src/util/image/MapCreator.java [hunk 3]
```diff
@@ -544,7 +554,20 @@ private static String getValue(final String arg) {
...
+          | UnsupportedLookAndFeelException e) {
+        ClientLogger.logError("Look and feel could not be applied to the Map Creator", e);
+      } catch(ClassCastException e){
+        ClientLogger.logQuietly("Argument is not a Look and Feel");
+      }
+    }
+    
+    if (args.length == 1 && !laf) {
       String value;
       if (args[0].startsWith(TRIPLEA_MAP_FOLDER)) {
```
Comment (DanVanAtta) [inline src/util/image/MapCreator.java]: Yeah, this is getting dirty.. If the first arg is a valid look and feel class, we use it, otherwise we assume it to be a folder location?? Args should probably have flags for them so it's not dependent upon arg position and value.

**Comment by DanVanAtta** [comment]
@RoiEXLab  the arg handling is not very clean anyways. It's probably better to assume all calls into the map creator will be from the UI. Anything that handles command line only invocation I think should be considered as unused (AKA dead code - and there is only one thing to do with dead code)

**Comment by RoiEXLab** [comment]
Here you go.. I looked into some old open issues and I decided to do some test on CSS button styling for the future.. sometime more or less soon I will probably start converting JButtons into javaFX buttons...

**Comment by frigoref** [comment]
Review status: 0 of 2 files reviewed at latest revision, 3 unresolved discussions.

---

*[src/util/image/MapCreator.java, line 76 \[r1\]](https://reviewable.io:443/reviews/triplea-game/triplea/621#-KGCzjF-RSfRoJrt3ZdW-r1-76:-KGE7ZhM6xMh9tO75hMr:-1295703649) ([raw file](https://github.com/triplea-game/triplea/blob/b1c448ec7973d07e22be2f06a53b42491b8d8fd2/src/util/image/MapCreator.java#L76)):*
Please consider GameRunne2.setupLookAndFeel for this piece of code and reuse the method here instead.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/triplea-game/triplea/621)*

**Comment by DanVanAtta** [comment]
@frigoref  has a good suggestion. Otherwise this is looking pretty good to me.

**Comment by RoiEXLab** [comment]
Everything should run fine now...
